### ISSUE TITLE
Dialog never updates aria-hidden attribute

### ISSUE THREAD

**Issue by kfranqueiro** [material-components/material-components-web#3535]
### What MDC Web Version are you using?

Any

### What browser(s) is this bug affecting?

All

### What are the steps to reproduce the bug?

Go to any dialog demo page e.g. https://material-components.github.io/material-components-web-catalog/#/component/dialog

### What is the expected behavior?

The dialog root element should have `aria-hidden="true"` when closed and remove the attribute when opened

### What is the actual behavior?

The JS component doesn't update/sync `aria-hidden` at all

### Any other information you believe would be useful?

I noticed this while working on the dialog prototype branch for #3413, but this is already an existing issue. I'm aiming to fix it for new dialog.

**Comment by kfranqueiro** [comment]
Update: After discussion, it appears we should actually not need to maintain the aria-hidden attribute for Dialog at all, now that it uses `display: none`. So instead we'll remove any mentions of it from examples and tests.
